distsql/distsql.go

```diff
@@ -169,5 +170,35 @@ func canUseChunkRPC(ctx sessionctx.Context) bool {
 	if ctx.GetSessionVars().EnableStreaming {
 		return false
 	}
+	if !supportedAlignment() {
+		return false
+	}
 	return true
 }
+
+// supportedAlignment checks the alignment in current system environment.
+// The alignment is influenced by system, machine and Golang version.
+// Using this function can guarantee the alignment is we want.
+func supportedAlignment() bool {
+	if unsafe.Sizeof(types.MysqlTime{}) != 16 ||
+		unsafe.Sizeof(types.Time{}) != 20 ||
+		unsafe.Sizeof(types.MyDecimal{}) != 40 {
+		return false
+	}
+	return true
+}
+
+// SetSystemEndian sets the system endian for the DAGRequest.
+func SetSystemEndian(dagReq *tipb.DAGRequest) {
+	dagReq.TidbSystemEndian = GetSystemEndian()
```

Comment:
We may check this when process starting and cache the result, instead of checking in each session.